pkg/common/security/acl_test.go

```diff
@@ -19,119 +19,206 @@
+	}
+	for _, tt := range tests {
+		t.Run(tt.caseName, func(t *testing.T) {
+			if _, err := NewACL(tt.acl); err == nil {
+				t.Errorf("parsing %s string should be failed", tt.acl)
+			}
+		})
 	}
 }
 
 func TestACLAccess(t *testing.T) {
-	acl, err := NewACL("user1,user2 group1,group2")
-	assert.NilError(t, err, "parsing failed for string: 'user1,user2 group1,group2'")
-	user := UserGroup{User: "", Groups: nil}
-	assert.Assert(t, !acl.CheckAccess(user), "no user, should have been denied")
-	user = UserGroup{User: "user1", Groups: nil}
-	assert.Assert(t, acl.CheckAccess(user), "user1 (no groups) should have been allowed")
-	user = UserGroup{User: "user3", Groups: []string{"group1"}}
-	assert.Assert(t, acl.CheckAccess(user), "group1 should have been allowed")
-	user = UserGroup{User: "user3", Groups: []string{"group3", "group1"}}
-	assert.Assert(t, acl.CheckAccess(user), "group1 (2nd group) should have been allowed")
-	user = UserGroup{User: "user3", Groups: []string{"group3"}}
-	assert.Assert(t, !acl.CheckAccess(user), "user3/group3 should have been denied")
-
-	acl, err = NewACL("*")
-	assert.NilError(t, err, "parsing failed for string: '*'")
-	user = UserGroup{User: "", Groups: nil}
-	assert.Assert(t, acl.CheckAccess(user), "no user, wildcard should have been allowed")
-	user = UserGroup{User: "user1", Groups: []string{"group1"}}
-	assert.Assert(t, acl.CheckAccess(user), "user1/group1, wildcard should have been allowed")
-
-	acl, err = NewACL("")
-	assert.NilError(t, err, "parsing failed for string: ''")
-	user = UserGroup{User: "", Groups: nil}
-	assert.Assert(t, !acl.CheckAccess(user), "no user, empty ACL always deny")
-	user = UserGroup{User: "user1", Groups: []string{"group1"}}
-	assert.Assert(t, !acl.CheckAccess(user), "user1/group1, empty ACL always deny")
+	tests := []struct {
+		acl      string
+		visitor  UserGroup
+		expected bool
+	}{
+		{
+			"user1,user2 group1,group2",
+			UserGroup{User: "", Groups: nil},
+			false,
+		},
+		{
+			"user1,user2 group1,group2",
+			UserGroup{User: "user1", Groups: nil},
+			true,
+		},
+		{
+			"user1,user2 group1,group2",
+			UserGroup{User: "user3", Groups: []string{"group1"}},
+			true,
+		},
+		{
+			"user1,user2 group1,group2",
+			UserGroup{User: "user3", Groups: []string{"group3", "group1"}},
+			true,
+		},
+		{
+			"user1,user2 group1,group2",
+			UserGroup{User: "user3", Groups: []string{"group3"}},
+			false,
+		},
+		{
+			"*",
+			UserGroup{User: "", Groups: nil},
+			true,
+		},
+		{
+			"*",
+			UserGroup{User: "user1", Groups: []string{"group1"}},
+			true,
+		},
+		{
+			acl:      "",
+			visitor:  UserGroup{User: "", Groups: nil},
+			expected: false,
+		},
+		{
+			"",
+			UserGroup{User: "user1", Groups: []string{"group1"}},
+			false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(fmt.Sprintf("vistor %v, acl %s", tt.visitor, tt.acl), func(t *testing.T) {
+			acl, err := NewACL(tt.acl)
+			if err != nil {
+				t.Error(err.Error())
```

Comment:
nit: explain that parse error is unexpected as you did before

Reply:
I would remove the err later because the acl string is in the create unit tests.

Reply:
Oops, I forgot that lint would request developer to handle the err in the UT.
So I updated the message of newACL error in the latest commit.